packages/@lwc/wire-service/src/link-context-event.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+/**
+ * Event fired by wire adapters to link to a context provider
+ */
+export class LinkContextEvent {
+    type: string;
+    uid: string;
```

Comment:
we can also bikeshed on the name of this unique id, which is really a DOM event name used to connect to the provider under the hood.